src/realm/object-store/object_accessor.hpp

```diff
@@ -354,6 +361,10 @@ Object Object::create(ContextType& ctx, std::shared_ptr<Realm> const& realm, Obj
         if (v)
             object.set_property_value_impl(ctx, prop, *v, policy, is_default);
     }
+    if (object_schema.is_asymmetric) {
+        obj.remove();
+        return Object{};
+    }
```

Comment:
We'll need similar logic when sync instructions are replayed as part of client reset with recovery.

Reply:
Will deleting the object not make the server delete it too?

Reply:
No